python/nav/statemon/checker/MysqlChecker.py

```diff
@@ -69,25 +76,22 @@ class MysqlConnection(object):
     def __init__(self, addr, timeout=None):
         host, _port = addr
         sock = socket.create_connection(addr, timeout)
-        self.file = sock.makefile('rw')
+        self.file = sock.makefile('rwb')
 
         self.seqno = 0
 
     def read_packet(self):
         header = self.file.read(4)
 
-        lll = ord(header[0])
-        mmm = ord(header[1])
-        hhh = ord(header[2])
+        lll, mmm, hhh, seqno = struct.unpack('BBBB', header)
```

Comment:
Bonus: easier to read

Reply:
If only there were a struct format to read a 24-bit integer 😛